File: kos_repo_analysis_export/code/backend_gateway/klf/intent_registry.py
registry = {}

def register_intent(intent: str, fn):
    if intent in registry:
        raise Exception(f"Intent {intent} already registered")
    registry[intent] = fn

def get_handler(intent: str):
    return registry.get(intent)

def register_core_handlers():
    register_intent("agent.ping", lambda p, m: {"pong": True})
    register_intent("agent.echo", lambda p, m: {"echo": p}) 

File: agents/kos_repo_analysis_claude/kos_repo_analysis_export/code/klf_core/intent-registry.ts
type HandlerFn = (payload: any, meta: any) => Promise<any>;

const registry: Record<string, HandlerFn> = {};

export function registerIntent(intent: string, handler: HandlerFn) {
  if (registry[intent]) {
    throw new Error(`Intent "${intent}" already registered`);
  }
  registry[intent] = handler;
}

export function getHandler(intent: string): HandlerFn | undefined {
  return registry[intent];
}

export function getAllIntents(): string[] {
  return Object.keys(registry);
} 